Make usage of `foo` clearer

bluesmoon opened the pull request:
This change replaces the term `foo` in the code example with `norm` so that the example works when entered into a julia runtime.  It also explains the requirements for the function.

Thanks to Avik Sengupta for the text and for all the commenters on https://discourse.julialang.org/t/what-does-foo-in-the-parallel-computing-docs-refer-to/4182 for making this happen.

bluesmoon committed:
Make usage of `foo` clearer

This change makes it clearer to new readers what the term `foo` in the code example refers to.

Thanks to Avik Sengupta for the text and for all the commenters on https://discourse.julialang.org/t/what-does-foo-in-the-parallel-computing-docs-refer-to/4182 for making this happen.

JeffBezanson commented:
Thanks! Maybe we can take this one step further and replace `foo` with `norm` for this whole example? Always better to use a meaningful function instead of `foo` if possible.

bluesmoon committed:
Make examples work when executed

This change replaces the placeholder `foo` with the built-in `norm` to make sure the example works when typed into a julia program.

bluesmoon commented:
@JeffBezanson updated accordingly.

Sacha0 commented on doc/src/manual/parallel-computing.md:
````diff
@@ -258,11 +258,15 @@ snippet:
+remotecall_fetch(()->norm(A), 2)
 ```
 
+In this case [`norm`](@ref) is a function that takes 2D array as a parameter, and MUST be defined in
````
With the change from `foo` to `norm`, perhaps this additional paragraph is no longer necessary?

bluesmoon replied:
As someone who found the docs confusing, I think it is necessary.  Someone might think that `norm` is the only function one can use to push a global to a worker.

ararslan commented on doc/src/manual/parallel-computing.md:
````diff
@@ -258,11 +258,15 @@ snippet:
 ```
 
+In this case [`norm`](@ref) is a function that takes 2D array as a parameter, and MUST be defined in
+the remote process.  You could, any function other than `norm` as long as it is defined in the remote
````
Do you mean
> You could **use** any function [...]

?

bluesmoon replied:
yes, fixed

bluesmoon committed:
Add missing word
